modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -484,7 +485,9 @@ public class IgniteImpl implements Ignite {
                 clock
         );
 
-        LongSupplier partitionIdleSafeTimePropagationPeriodMsSupplier = partitionIdleSafeTimePropagationPeriodMsSupplier();
+        ReplicationConfiguration replicationConfig = clusterConfigRegistry.getConfiguration(ReplicationConfiguration.KEY);
+
+        LongSupplier partitionIdleSafeTimePropagationPeriodMsSupplier = partitionIdleSafeTimePropagationPeriodMsSupplier(replicationConfig);
```

Comment:
What's the point of having suppliers here?

Reply:
First it was because configuration object seems to be too complex to pass when you just need to get one property. This is arguable.

But then it turned out that it would be cool to verify from the outside (wrt configuration) that the configuration returns the same value. Also, this logic is needed in a few components, so it seems natural to have a common supplier.
We can monitor the situation and later remove the assertions, probably returning to operating vanilla configuration objects.